extensions/amp-font/0.1/amp-font.js

```diff
@@ -0,0 +1,327 @@
+    containerElement.appendChild(customFontElement);
+    this.document_.body.appendChild(containerElement);
+  }
+
+
+  /**
+   * Compare dimensions between elements styled with default fonts and custom
+   * font.
+   * @returns {boolean} Returns true if the dimensions are noticeably different
+   * else returns false.
+   * @private
+   */
+  compareMeasurements_() {
+    let hasWidthChanged = false;
+    let hasHeightChanged = false;
+    this.domElements_.DEFAULT_FONT_ELEMENTS.some(defaultElement => {
+      hasWidthChanged = (
+          Math.abs(
+              defaultElement./*REVIEW*/offsetWidth -
+              this.domElements_.CUSTOM_FONT_ELEMENT./*REVIEW*/offsetWidth) >
+            TOLERANCE_);
+      hasWidthChanged = (
+          Math.abs(
+              defaultElement./*REVIEW*/offsetHeight -
+              this.domElements_.CUSTOM_FONT_ELEMENT./*REVIEW*/offsetHeight) >
+            TOLERANCE_);
+      if (hasWidthChanged || hasHeightChanged) {
+        return true;
+      }
+    });
+    if (hasWidthChanged || hasHeightChanged) {
+      return true;
+    }
+    return false;
+  }
+
+
+  /**
+   * @private
+   */
+  onFontLoadSuccess_() {
+    timer.cancel(this.timeoutHandle_);
+    var addClassName = this.element.getAttribute('on-load-add-class');
+    var removeClassName =
+        this.element.getAttribute('on-load-remove-class');
+    if (addClassName) {
+      this.documentElement_.classList.add(addClassName);
+    }
+    if (removeClassName) {
+      this.documentElement_.classList.remove(removeClassName);
+      this.document_.body.classList.remove(removeClassName);
+    }
+    this.dispose_();
+  }
+
+
+  /**
+   * @private
+   */
+  onFontLoadError_() {
+    var addClassName = this.element.getAttribute('on-error-add-class');
+    var removeClassName =
+        this.element.getAttribute('on-error-remove-class');
+    if (addClassName) {
+      this.documentElement_.classList.add(addClassName);
+    }
+    if (removeClassName) {
+      this.documentElement_.classList.remove(removeClassName);
+      this.document_.body.classList.remove(removeClassName);
+    };
+    this.dispose_();
+  }
+
+
+  /**
+   * @private
+   */
+  dispose_() {
+    if (this.domElements_.CONTAINER) {
+      removeElement(this.domElements_.CONTAINER);
+    }
+    this.domElements_ = {
+      CONTAINER: null,
+      DEFAULT_FONT_ELEMENTS: [],
+      CUSTOM_FONT_ELEMENT: null
+    };
+    if (this.timeoutHandle_) {
+      timer.cancel(this.timeoutHandle_);
+    }
+  }
+
+
+  /**
+   * Starts the timeout counter to not attempt loading the font after the
+   * timeout normal.
+   * @private
+   */
+  beginTimeout_() {
+    let timeoutAttribute = this.element.getAttribute('timeout');
+    let timeoutInMs = parseInt(timeoutAttribute, 10);
+    timeoutInMs = isNaN(timeoutInMs) || timeoutInMs < 0 ?
+        DEFAULT_TIMEOUT_ : timeoutInMs;
+    let timeSinceStart = timer.now() - this.startTime_;
+    timeoutInMs = timeSinceStart > timeoutInMs ?
```

Comment:
So, if as the result we end up with `timeoutInMs == 0` - we should still wait some time - we can't just wait for `0` in this case since it gives us no guarantees that font integration in browser would be this fast. For the user `timeout=0` has a clear meaning - "Don't wait for download; only use font when in cache". However, from a browser's point of view `0` doesn't mean this. So, what we need to do instead is use the expression:
```
timeoutInMs = Math.max(timeoutInMs, CACHED_FONT_LOAD_TIME_);
```
CACHED_FONT_LOAD_TIME_ is a const. You have to experiment a little to find out what this is. I expect it'd be at least 100-200ms.